### DISCUSSION TITLE
When you set a node to .visible = false, its child nodes are no longer visible

### DISCUSSION THREAD

**Discussion by gomezleo35** [google/model-viewer#4461] [Q&A]
Hi

Description of the problem

I have experienced this error with a model .glb.
It happens that when I hide a node, all the child nodes disappear. Regardless of whether each contains its visible attribute with the value true. All of its child nodes, or the other nodes in the model, still have their attribute visible with the value set to true, but they can no longer be displayed.
On the other hand, if the parent node has its visible attribute with a value of “true”, the child nodes can be hidden and shown without any problem as if they behaved in groups.
Materials can't share visibility, right?
What could be generating this error?
Does anyone know a solution for this?

Thank you

**Comment by danilokrulj** [comment]
Honestly that doesn't sound like an error to me, I think that is the expected behaviour. If you hide a parent node its children should be hidden too I suppose.

**Reply by gomezleo35** [reply]
But it behaves this way only for a specific node and a specific glb model.
The same does not happen for the rest of the .glb models.
It was verified that each child node maintains its visible attribute at "true" when the visible attribute of the parent is changed.

**Reply by elalish** [reply]
Critically, `model-viewer` has no visibility API, so you must be hacking down to the three.js layer underneath. If you want help with three.js, please ask on their forums.
